File: .leetcode/3.无重复字符的最长子串.c
/*
 * @lc app=leetcode.cn id=3 lang=c
 *
 * [3] 无重复字符的最长子串
 */

// @lc code=start
int lengthOfLongestSubstring(char* s) {
    int max_len=0;
    int current_len=0;
    int s_len=strlen(s);
    char* t=malloc(sizeof(char)*s_len);
    for(int left=0;left<s_len;left++){
        for(int i=left;i>=left-current_len;i--){
            if(s[left]==s[i])current_len=0; //有一样的，current_len重新计数
        }
        current_len++;
        if(current_len>max_len)max_len=current_len;
    }
    return max_len;
}
// @lc code=end

File: .leetcode/3.无重复字符的最长子串.py
#
# @lc app=leetcode.cn id=3 lang=python
#
# [3] 无重复字符的最长子串
#

# @lc code=start
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        n=len(s)
        if n<=1:
            return n
            
        max_len=1
        for left in range(n):
            ha=[0]*1000
            ha[ord(s[left])]+=1
            right=left+1
            while right<n and ha[ord(s[right])]==0:
                ha[ord(s[right])]+=1  
                right+=1

            max_len=max(max_len,right-left)

        return max_len
        
# @lc code=end

